lib/main.js

```diff
@@ -112,67 +162,76 @@ export default {
     tslintDef = require('loophole').allowUnsafeNewFunction(() => require(TSLINT_MODULE_NAME).Linter);
 
     return {
+      name: 'TSLint',
       grammarScopes: ['source.ts', 'source.tsx'],
       scope: 'file',
       lintOnFly: true,
-      lint: (textEditor) => {
-        const filePath = textEditor.getPath();
-        const text = textEditor.getText();
-
-        return this.getLinter(filePath).then((Linter) => {
-          const configurationPath = Linter.findConfigurationPath(null, filePath);
-          const configuration = Linter.loadConfigurationFromPath(configurationPath);
-
-          let { rulesDirectory } = configuration;
-          if (rulesDirectory) {
-            const configurationDir = path.dirname(configurationPath);
-            if (!Array.isArray(rulesDirectory)) {
-              rulesDirectory = [rulesDirectory];
-            }
-            rulesDirectory = rulesDirectory.map((dir) => {
-              if (path.isAbsolute(dir)) {
-                return dir;
-              }
-              return path.join(configurationDir, dir);
-            });
-
-            if (this.rulesDirectory) {
-              rulesDirectory.push(this.rulesDirectory);
-            }
-          }
+      lint: async (textEditor) => {
+        if (this.ignoreTypings && textEditor.getPath().toLowerCase().endsWith('.d.ts')) {
+          return [];
+        }
 
-          const linter = new Linter({
-            formatter: 'json',
-            rulesDirectory,
-          });
+        return this.lint(textEditor);
+      },
+    };
+  },
 
-          linter.lint(filePath, text, configuration);
-          const lintResult = linter.getResult();
+  lint(textEditor, options) {
+    const filePath = textEditor.getPath();
+    const text = textEditor.getText();
 
-          if (textEditor.getText() !== text) {
-            // Text has been modified since the lint was triggered, tell linter not to update
-            return null;
-          }
+    return this.getLinter(filePath).then((Linter) => {
+      const configurationPath = Linter.findConfigurationPath(null, filePath);
+      const configuration = Linter.loadConfigurationFromPath(configurationPath);
 
-          if (!lintResult.failureCount) {
-            return [];
+      let { rulesDirectory } = configuration;
+      if (rulesDirectory) {
+        const configurationDir = path.dirname(configurationPath);
+        if (!Array.isArray(rulesDirectory)) {
```

Comment:
I have no idea, this code was here before my change and is just moving with my refactoring of the method